node/test/hyperbahn/hostports.js

```diff
@@ -0,0 +1,267 @@
+            assert.equals(res.body.message, 'no peer available for matt', 'error message as expected');
+            assert.end();
+        }
+    });
+
+    HyperbahnCluster.test('get host port as expected', {
+        size: 5
+    }, function t(cluster, assert) {
+        var bob = cluster.remotes.bob;
+        var steve = cluster.remotes.steve;
+        var steveSub = steve.channel.makeSubChannel({
+            serviceName: 'hyperbahn',
+            peers: cluster.hostPortList
+        });
+        var request = steveSub.request({
+            headers: {
+                cn: 'test'
+            },
+            serviceName: 'hyperbahn',
+            hasNoParent: true
+        });
+
+        var client = new HyperbahnClient({
+            serviceName: 'hello-bob',
+            callerName: 'hello-bob-test',
+            hostPortList: cluster.hostPortList,
+            tchannel: bob.channel,
+            logger: DebugLogtron('hyperbahnClient')
+        });
+
+        client.once('advertised', onResponse);
+        client.advertise();
+
+        function onResponse() {
+            thrift.send(request,
+                'Hyperbahn::discover',
+                null,
+                {
+                    query: {
+                        serviceName: 'hello-bob'
+                    }
+                },
+                check
+            );
+        }
+
+        function check(err, res) {
+            if (err) {
+                assert.end(err);
+            }
+            assert.ok(res, 'should be a result');
+            assert.ok(res.ok, 'result should be ok');
+            console.log(res.body.peers[0]);
+            assert.equals(covertHost(res.body.peers[0]), bob.channel.hostPort.replace('localhost', '127.0.0.1'),
+                'should get the expected hostPort');
+            client.destroy();
+            assert.end();
+        }
+    });
+
+    HyperbahnCluster.test('malformed thrift IDL: empty serviceName', {
+        size: 5
+    }, function t(cluster, assert) {
+        var bob = cluster.remotes.bob;
+        var bobSub = bob.channel.makeSubChannel({
+            serviceName: 'hyperbahn',
+            peers: cluster.hostPortList
+        });
+        var request = bobSub.request({
+            headers: {
+                cn: 'test'
+            },
+            serviceName: 'hyperbahn',
+            hasNoParent: true
+        });
+        thrift.send(request,
+            'Hyperbahn::discover',
+            null,
+            {
+                query: {
+                    serviceName: ''
+                }
+            },
+            onResponse
+        );
+        function onResponse(err, res) {
+            if (err) {
+                assert.end(err);
+            }
+            assert.ok(!res.ok, 'should be not ok');
+            // TODO: the res.body.message is missing, need to assert on that
+            assert.end();
+        }
+    });
+
+    // HyperbahnCluster.test('malformed thrift IDL: an empty body', {
```

Comment:
These are minor issues of thriftrw and only occur under bad IDLs (very uncommon scenarios). They are also orthogonal to this PR. Can we land this PR first? :)